azavea/echo-locator
# Replace Amplify authentication with anonymous-only login

## Overview

This PR completely removes Amplify and all of its related frontend authentication components and replaces the login flow with anonymous log in by replacing the `Authentication` HOC with a smaller `anonymousAuthenticator` HOC.

### Demo

![login_AdobeCreativeCloudExpress](https://user-images.githubusercontent.com/77936689/162217628-4b89bb31-68aa-49dc-8fdb-1ac479beb320.gif)

You are now only able to log in anonymously; doing so will take you directly to fill out your profile information.

### Notes

I had a tough time deciding the extend of what to cut out versus leave in for the sake of building from for Django authentication. Ultimately, along with removing anything that was explicitly related to Amplify's authentication process, I chose to remove places where currently useless checks for anonymity and/or authentication level impacted code readability prior to the addition of Django authentication, but left them in where they could be helpful for future authentication but not distracting. An example of this is removing the checks and related code for `isAnonymous` and `isCounselor` in `edit-profile.js`, as an anonymous user should never be presented with the option to re-send their invitation email, but leaving `clientAccountConfirmed`, `clientEmail`, and `clientInviteSent` as properties on the `AccountProfile` type. Following this rule, I have removed all checks for `isAnonymous`, `isCounselor`, and account validation, but have left `EMAIL_REGEX` in `constants.js` and email-related strings in the translation files.

One strange behavior I found following these changes was if you clicked "Continue without an account" and then clicked the browser's back button before completing the form, it would take you to a blank "search" page, rather than returning to the login page, unless you cleared your browser's local storage. This was compared to `develop`, where clicking back after selecting anonymous login takes you back to wherever you navigated to the login page from. I fixed this by replacing the redirect ternaries in `application.js` to redirect to '/profile' instead of '/search'. This fix is a little odd in the case of the 3rd path to '/profile', but it seemed a decent placeholder until we restore authentication.

## Testing Instructions

 * Run `./scripts/setup`
 * Run `./scripts/server` and navigate to **http://localhost:9966/**
 * Select "Continue without an account". It should take you to the profile page. 
 * Click your browser's back button and verify that it doesn't take you to a blank search page. ~~Then fill out the form and click save.~~
 * Click Cancel and verify that it takes you back to the login page. Select "Continue without an account" again and fill out the form and click save.
 * Verify that you can favorite neighborhoods, make changes to your profile, and that your profile is saved on browser refresh.

EDIT: Added testing instructions for re-route to login on cancel during profile edit.


Resolves #431
